audformat/core/utils.py

```diff
@@ -1622,8 +1625,18 @@ def job(file: str) -> pd.Timedelta:
                 progress_bar=verbose,
                 task_description='Read duration',
             )
-            ends.values[idx_nat] = durs
 
+            # Now replace all NaT entries in end
+            # by the collected duration values.
+            # We have to convert ends to a series first
+            # in order to preserve precision of duration values
+
+            ends = ends.to_series()
+            ends[idx_nat] = durs
+
+            # Create a new index
+            # as index.set_levels() does not work
+            # if the level contains only NaT entries
```

Comment:
When can this happen? I thought we just replaced all NaTs.

Reply:
You are right, this can no longer happen.
I tried to replace creating a new index with
```python
index = index.set_levels(ends, level=2)
```
but this fails for examples with files having the same duration with
```
ValueError: Level values must be unique: [Timedelta('0 days 00:00:01'), Timedelta('0 days 00:00:01')] on level 2
```

Should I stay with the current implementation, but just update the comment?

Reply:
As far as I know, with `set_levels()` you can replace the existing values, but you cannot have more values afterwards. But that's what we do here as we replace a single `NaT` entry with multiple different values. I guess simply removing the comment should be fine.

Reply:
Thanks, I also figured out why `set_levels()` does not work in the mean time, and as there is no `set_level_values()` that we could use instead I decided to stay with the current implementation and removed the comment

Reply:
The same problem was discussed at https://github.com/pandas-dev/pandas/issues/28294#issuecomment-528456052 and there it was also recommended to create a new index.